apps/api/src/domain/models/vocabulary-list/commands/add-term-to-vocabulary-list/index-term-added-to-vocabulary-list-on-term-view.event-handler.integration.spec.ts

```diff
@@ -0,0 +1,150 @@
+
+const vocabularyListNameInEnglish = 'vl name (english)';
+
+const vocabularyListCompositeId = {
+    type: AggregateType.vocabularyList,
+    id: vocabularyListId,
+};
+
+const termId = buildDummyUuid(1);
+
+const existingTermView = buildTestInstance(TermViewModel);
+
+const vocabularyListView = buildTestInstance(VocabularyListViewModel, {
+    id: vocabularyListId,
+    name: buildMultilingualTextFromBilingualText(
+        { text: vocabularyListName, languageCode: originalLanguageCode },
+        { text: vocabularyListNameInEnglish, languageCode: translationLanguageCode }
+    ),
+    entries: [],
+});
+
+const termAddedEvent = new TestEventStream().buildSingle<TermAddedToVocabularyList>({
+    type: 'TERM_ADDED_TO_VOCABULARY_LIST',
+    payload: {
+        aggregateCompositeIdentifier: vocabularyListCompositeId,
+        termId,
+    },
+});
+
+describe(`IndexTermAddedToVocabularyListEventHandlerOnTermView.handle`, () => {
+    let testVocabularyListQueryRepository: IVocabularyListQueryRepository;
+
+    let termQueryRepository: ITermQueryRepository;
+
+    let databaseProvider: ArangoDatabaseProvider;
+
+    let app: INestApplication;
+
+    let eventHandler: IndexTermAddedToVocabularyListOnTermViewEventHandler;
+
+    beforeAll(async () => {
+        const moduleRef = await Test.createTestingModule({
+            imports: [PersistenceModule.forRootAsync()],
+        })
+            .overrideProvider(ConfigService)
+            .useValue(
+                buildMockConfigService(
+                    {
+                        ARANGO_DB_NAME: generateDatabaseNameForTestSuite(),
+                    },
+                    buildConfigFilePath(Environment.test)
+                )
+            )
+            .compile();
+
+        await moduleRef.init();
+
+        app = moduleRef.createNestApplication();
+
+        const connectionProvider = app.get(ArangoConnectionProvider);
+
+        databaseProvider = new ArangoDatabaseProvider(connectionProvider);
+
+        testVocabularyListQueryRepository = new ArangoVocabularyListQueryRepository(
+            connectionProvider,
+            new ConsoleCoscradCliLogger()
+        );
+
+        const audioRepository = new ArangoAudioItemQueryRepository(connectionProvider);
+
+        termQueryRepository = new ArangoTermQueryRepository(
+            connectionProvider,
+            audioRepository,
+            new ConsoleCoscradCliLogger()
+        );
+
+        eventHandler = new IndexTermAddedToVocabularyListOnTermViewEventHandler(
+            termQueryRepository,
+            testVocabularyListQueryRepository
+        );
+    });
+
+    afterAll(async () => {
+        databaseProvider.close();
+    });
+
+    beforeEach(async () => {
+        await databaseProvider.clearViews();
+
+        await testVocabularyListQueryRepository.create(vocabularyListView);
+
+        await termQueryRepository.create(existingTermView);
+    });
+
+    describe(`when there is an existing vocabulary list`, () => {
```

Comment:
Is there a test case for when the vocabulary list some how does not exist? It should fail gracefully.

Reply:
I see that below now.